### PR TITLE
<FEAT>: Add in partial deployment functionality

### PR DESCRIPTION
**Pull request by SEAJang**
# Pull Request Key Information

## Reason for change

Some work practices may only be on a subset of sites.

## What was changed

- add in a new parameter to the infrastructure to allow for partial deployment
- Update unit test
- Fix bug related to propagating parameter only working for when site type is present.

## Intended Purpose

This change allows for the user to define a subset of sites to deploy methods to.

## Testing Completed

All unit tests pass
[results.txt](https://github.com/LDAR-Sim/LDAR_Sim/files/15504215/results.txt)

Ran manual simulations with the simple_test_case1 - ran the normal and then updated the infrastructure with the following changes:
Setting site deployment for OGI to false on all sites through the site file
[facilities_Perm.csv](https://github.com/LDAR-Sim/LDAR_Sim/files/15503752/facilities_Perm.csv)
- Checked that there was no mitigation done by P_OGI

Setting the Site deployment for OGI based on site type - all set to FALSE
[site_type.csv](https://github.com/LDAR-Sim/LDAR_Sim/files/15504104/site_type.csv)
- Checked that there was no mitigation done by P_OGI

A site type set to True.
[site_type.csv](https://github.com/LDAR-Sim/LDAR_Sim/files/15504114/site_type.csv)
- Checked that there was only a fraction of surveys going based on the site type

### CODE REVIEW

File: LDAR_Sim/src/parameters/high_level_parameters.py [hunk 1]
```diff
@@ -19,7 +19,8 @@
 #
 # ------------------------------------------------------------------------------
 
-from typing import Any, override
+from typing import Any
+from typing_extensions import override
```
Comment (ThomasGalesloot) [inline LDAR_Sim/src/parameters/high_level_parameters.py:23]: Why was this change necessary?

File: LDAR_Sim/src/scheduling/generic_schedule.py [hunk 2]
```diff
@@ -65,7 +65,8 @@ def _set_survey_plans(
             # TODO flush out what survey planner needs as inputs for the constructor
             survey_freq: int = site._survey_frequencies[self._method]
             # TODO: make sure this is in the correct place....
-            if survey_freq is None:
+            deploy_meth: bool = site.check_site_deployable(self._method)
```
Comment (ThomasGalesloot) [inline LDAR_Sim/src/scheduling/generic_schedule.py:68]: Nit; Can we rename this to something like do_site_deployment? Worried deploy meth might be confused later to be the name of the method

File: LDAR_Sim/src/virtual_world/infrastructure.py [hunk 3]
```diff
@@ -118,18 +126,18 @@ def update_propagating_params(
                     if site_type_val is not None:
                         prop_params[pdc.Common_Params.METH_SPECIFIC][param][method] = site_type_val
 
-            # Updating propagating parameters with site info
-            for param in IC.Sites_File_Constants.PROPAGATING_PARAMS:
-                site_val = site_row_df_info.get(param, None)
-                if site_val is not None:
-                    prop_params[param] = site_val
+        # Updating propagating parameters with site info
```
Comment (ThomasGalesloot) [inline LDAR_Sim/src/virtual_world/infrastructure.py:129]: Why did the indentation change here?

File: LDAR_Sim/testing/unit_testing/test_virtual_world/test_infrastructure/test_update_propagating_params.py [hunk 4]
```diff
@@ -48,6 +48,13 @@ def generate_test_prop_params(draw, methods):
         to_ret[cp.METH_SPECIFIC][key] = {}
         for method in methods:
             to_ret[cp.METH_SPECIFIC][key][method] = draw(st.integers(min_value=0, max_value=10000))
+    to_ret[cp.METH_SPECIFIC][
+        Infrastructure_Constants.Sites_File_Constants.SITE_DEPLOYMENT_PLACEHOLDER
+    ] = {}
+    for method in methods:
+        to_ret[cp.METH_SPECIFIC][
+            Infrastructure_Constants.Sites_File_Constants.SITE_DEPLOYMENT_PLACEHOLDER
+        ][method] = {}
```
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_virtual_world/test_infrastructure/test_update_propagating_params.py:57]: Is this supposed to be a dictionary?

**Review by ThomasGalesloot — COMMENTED** [review]
Great work - While not necessary it would be good to see one more manual test, with a random half of the sites in the sites file with no deployment and the other half with deployment.

File: LDAR_Sim/src/parameters/high_level_parameters.py (hunk 1 above)
Comment (SEAJang) [reply LDAR_Sim/src/parameters/high_level_parameters.py:23]: It's not using the correct import, and was failing to run.

File: LDAR_Sim/testing/unit_testing/test_virtual_world/test_infrastructure/test_update_propagating_params.py (hunk 4 above)
Comment (SEAJang) [reply LDAR_Sim/testing/unit_testing/test_virtual_world/test_infrastructure/test_update_propagating_params.py:57]: Yes, it makes a dictionary with the method names as keys, then the actual values needed.

File: LDAR_Sim/src/virtual_world/infrastructure.py (hunk 3 above)
Comment (SEAJang) [reply LDAR_Sim/src/virtual_world/infrastructure.py:129]: This was a bug fix. 
The sites file specific propagating parameters should always happen. They shouldn't be limited by if the site type file exists.

**Comment by SEAJang** [comment]
[results.txt](https://github.com/user-attachments/files/15686269/results.txt)
All unit tests pass - Unit test results after the function rename. 

Manual testing with the following infrastructure file resulted in the expected partial deployment.
[facilities_Perm.csv](https://github.com/user-attachments/files/15686368/facilities_Perm.csv)
